### PR TITLE
feat(notifications): rewrite notification settings interface

### PR DESCRIPTION
**Pull request by hypeJunction**
Notifications plugin has been rewritten for improved usability, extendability and scalability.

BREAKING CHANGE
Notifications plugin has been rewritten. Plugins that rely on views, actions and styles of the notifiations plugin have to be updated.

joxi_screenshot_1475077452658

joxi_screenshot_1475077480940

joxi_screenshot_1475077507092

### CODE REVIEW

**Comment by hypeJunction** [comment]
I don't care for the icons and would be very happy to get rid of them - that would simplify both the JS and CSS.

**Review by jdalsem — APPROVED** [review]

**Comment by hypeJunction** [comment]
@Elgg/contributors Cast your vote https://elgg.org/discussion/view/2551749/support-for-notification-settings-for-friend-collections

**Comment by hypeJunction** [comment]
This is now scalable. Friendspicker out of the notifications interface.

File: docs/guides/upgrading.rst [hunk 1]
```diff
@@ -22,6 +22,13 @@ Removed views
 
  * ``resources/file/download``
  * ``input/write_access``: mod/pages now uses the **access:collections:write** plugin hook.
+ * ``subscriptions/form/additions``: extend ``forms/notificationsettings/save`` instead
```
Comment (hypeJunction) [inline docs/guides/upgrading.rst]: Not the case anymore. Should use `subscriptions/settings/other` or whatever I named it an extension view.

**Comment by hypeJunction** [comment]
Added new screenshots.

File: mod/notifications/views/default/notifications/subscriptions/users.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * Friends and friend collection preferences
```
Comment (hypeJunction) [inline mod/notifications/views/default/notifications/subscriptions/users.php]: Just friends

File: mod/notifications/views/default/notifications/subscriptions/collections.php [hunk 3]
```diff
@@ -1,182 +0,0 @@
...
-/**
- * @uses $vars['user'] ElggUser
- */
-
-/* @var ElggUser $user */
-$user = elgg_extract('user', $vars);
-
-//@todo JS 1.8: no ?>
-<script>
-	function setCollection(members, method, id) {
```
Comment (mrclay) [inline mod/notifications/views/default/notifications/subscriptions/collections.php]: Great to see these views drenched in red.

Comment (hypeJunction) [inline mod/notifications/views/default/notifications/subscriptions/collections.php]: Yeah. Zero tolerance for spaghetti

**Comment by mrclay** [comment]
Replaced by #10340

File: mod/notifications/views/default/notifications/subscriptions/users.php (hunk 2 above)
Comment (mrclay) [inline mod/notifications/views/default/notifications/subscriptions/users.php]: Addressed in #10340

**Pull request closed without merging**